src/lib.rs

```diff
@@ -771,6 +799,47 @@ impl UnixDatagram {
         }
     }
 
+    /// Receives data on the socket.
+    ///
+    /// If path is None, the peer address set by the `connect` method will be used.  If it has not
+    /// been set, then this method will return an error.
+    ///
+    /// This interface allows sending data from multiple buffers.  This acts as if the buffers had been
+    /// concatenated in the order they were given.
+    ///
+    /// ctrl_msgs are special ancillary data that can be sent, such as file descriptors and Unix credentials
+    ///
+    /// flags is a pass-through of the flags specified in the sendmsg(2) man page
+    ///
+    /// On success, returns the number of bytes written.
+    pub fn recvmsg(&self, buffers: &[&mut[u8]], cmsg_buffer: &mut [u8], flags: libc::c_int) -> io::Result<RecvMsgResult> {
```

Comment:
The APIs here seem slightly too low-level, IMO, specifically the `flags` field here.